pynitrokey/nk3/utils.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2021 Nitrokey Developers
+#
+# Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+# http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+# http://opensource.org/licenses/MIT>, at your option. This file may not be
+# copied, modified, or distributed except according to those terms.
+
+from functools import total_ordering
+from typing import Tuple
+
+from spsdk.sbfile.misc import BcdVersion3
+
+
+@total_ordering
+class Version:
+    def __init__(self, major: int, minor: int, patch: int) -> None:
+        self.major = major
+        self.minor = minor
+        self.patch = patch
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Version):
+            return NotImplemented
+        return self._as_tuple() == other._as_tuple()
+
+    def __lt__(self, other: object) -> bool:
+        if not isinstance(other, Version):
+            return NotImplemented
+        return self._as_tuple() < other._as_tuple()
+
+    def __repr__(self) -> str:
+        return f"Version(major={self.major}, minor={self.minor}, patch={self.patch}"
+
+    def __str__(self) -> str:
+        return f"v{self.major}.{self.minor}.{self.patch}"
+
+    def _as_tuple(self) -> Tuple[int, int, int]:
+        return (self.major, self.minor, self.patch)
```

Comment:
I think this partly duplicates dataclass implementation (for the future):
- https://docs.python.org/3/library/dataclasses.html

Reply:
Good point, unfortunately requires Python 3.7.

Reply:
That's a shame indeed. We should survey the currently supported OSes with its Python distributions.
This could be a good starting point:
- https://repology.org/project/python/packages